lib/services/local/dynamic_desktops_test.go

```diff
@@ -0,0 +1,225 @@
+/**
+ * Teleport
+ * Copyright (C) 2024 Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package local
+
+import (
+	"context"
+	"github.com/google/go-cmp/cmp"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/backend"
+	"github.com/gravitational/teleport/lib/backend/memory"
+	"github.com/gravitational/trace"
+	"github.com/jonboulle/clockwork"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/protobuf/testing/protocmp"
+	"testing"
+)
+
+func newDynamicDesktop(t *testing.T, name string) types.DynamicWindowsDesktop {
+	desktop, err := types.NewDynamicWindowsDesktopV1(name, nil, types.DynamicWindowsDesktopSpecV1{
+		Addr: "xyz",
+	})
+	require.NoError(t, err)
+	return desktop
+}
+
+func setupDynamicDesktopTest(t *testing.T) (context.Context, *DynamicWindowsDesktopService) {
+	t.Parallel()
+	ctx := context.Background()
+	clock := clockwork.NewFakeClock()
+	mem, err := memory.New(memory.Config{
+		Context: ctx,
+		Clock:   clock,
+	})
+	require.NoError(t, err)
+	service := NewDynamicWindowsDesktopService(backend.NewSanitizer(mem))
+	return ctx, service
+}
+
+func TestDynamicWindowsService_CreateDynamicDesktop(t *testing.T) {
+	ctx, service := setupDynamicDesktopTest(t)
```

Comment:
Any reason to not include `t.Parallel()` here and everywhere below?

Reply:
It is included in `setupDynamicDesktopTest`